.github/workflows/neo4j-to-neptune-ci.yml

```
@@ -0,0 +1,92 @@
+name: Neo4j-to-Neptune CI
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - 'neo4j-to-neptune/**'
+    branches:
+      - master
+  push:
+    paths:
+      - 'neo4j-to-neptune/**'
+    branches:
+      - master
+
+jobs:
+  build-test-and-analyze:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up JDK 17
+      uses: actions/setup-java@v4
+      with:
+        java-version: '17'
+        distribution: 'corretto'
+
+    - name: Cache Maven dependencies
+      uses: actions/cache@v3
```

Nit: I think there's a version 4 available?

```suggestion
      uses: actions/cache@v4
```